azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -141,6 +142,7 @@ def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
             raise ProtocolError(msg="Error fetching goal state", inner=exception)
         finally:
             logger.info('Fetch goal state completed')
+            self.fetch_timestamp = datetime.utcnow()
```

Comment:
can we capture the time at which we fetch the actual goal state (in case there are delays for the other requests)

Reply:
Sure, let me change that.